pkg/services/provisioning/datasources/config_reader.go

```diff
@@ -62,7 +62,7 @@ func (cr *configReader) parseDatasourceConfig(path string, file os.FileInfo) (*D
 	}
 
 	if apiVersion.ApiVersion > 0 {
-		var v1 *DatasourcesAsConfigV1
+		v1 := &DatasourcesAsConfigV1{log: cr.log}
 		err = yaml.Unmarshal(yamlFile, &v1)
```

Comment:
Hmm interesting, you are right. I think it is possible yaml does some reflection magic traversing the pointer until it gets to something it can fill. Will fix that anyway.